gusto/equations/shallow_water_equations.py

```diff
@@ -272,7 +232,245 @@ def __init__(self, domain, parameters, fexpr=None, bexpr=None,
                          linearisation_map=linearisation_map,
                          u_transport_option=u_transport_option,
                          no_normal_flow_bc_ids=no_normal_flow_bc_ids,
-                         active_tracers=active_tracers, thermal=thermal)
+                         active_tracers=active_tracers)
+
+        # Use the underlying routine to do a first linearisation of the equations
+        self.linearise_equation_set()
+
+
+class ThermalShallowWaterEquations(ShallowWaterEquations):
+    u"""
+    Class for the (rotating) shallow-water equations, which evolve the velocity
+    'u' and the depth field 'D', via some variant of either:                  \n
+    ∂u/∂t + (u.∇)u + f×u + b*∇(D+B) + 0.5*D*∇b= 0,                            \n
+    ∂D/∂t + ∇.(D*u) = 0,                                                      \n
+    ∂b/∂t + u.∇(b) = 0,                                                       \n
+    for Coriolis parameter 'f', bottom surface 'B' and buoyancy field b,
+    or, if equivalent_buoyancy=True:
+    ∂u/∂t + (u.∇)u + f×u + b_e*∇(D+B) + 0.5*D*∇(b_e + beta_2 q_v)= 0,         \n
+    ∂D/∂t + ∇.(D*u) = 0,                                                      \n
+    ∂b_e/∂t + u.∇(b_e) = 0,                                                   \n
+    ∂q_t/∂t + u.∇(q_t) = 0,                                                   \n
```

Comment:
This matches the code below which requires us to have the `q_t` moisture variable when using the equivalent buoyancy formulation. I had imagined we would do it with an `ActiveTracer` object but this makes sense to me

Reply:
Yes, I wondered about how to do this... this way requires less work for the user and does the equivalent buoyancy formulation even mean anything without the moisture??

Reply:
Yes I agree with you both!